earth_enterprise/src/common/performancelogger_unittest.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright 2018 The Open GEE Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <string>
+#include <time.h>
+
+#include "performancelogger.h"
+
+using namespace std;
+using namespace testing;
+using namespace getime;
+
+// Mock of the PerformanceLogger class
+class MockPerformanceLogger {
```

Comment:
Not a blocker for this review - but is there a reason we don't subclass PerformanceLogger for the mock test, or have any explicit type safety?   Perhaps the output in PerformanceLogger (i.e., notify(NFY_NOTICE)  could be a separate method and overridden)

Reply:
Here's my thoughts on that. I feel that you get pretty good type safety with templates because the code won't compile if the class you provide to the template doesn't have the right functions. I could have created an abstract class to be the PerformanceLoggerInterface or something like that and coded to the interface, but that introduces another level of redirection and could be a performance problem on code like this that's called repeatedly.

Also, I intentionally am not testing the PerformanceLogger class in this PR because it only includes the log function and that is expected to change dramatically in follow up tickets, so I felt that those tests would be wasted.

Reply:
I came to a similar conclusion from looking at the previous code, that notify(...) could possibly be moved, it would require a number of changes, and perhaps is not the right path to look down, but it is one thing I looked into. This was on previous code though